cardinal/ecs/cql/cql.go

```diff
@@ -0,0 +1,135 @@
+package cql
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/alecthomas/participle/v2"
+)
+
+type cqlOperator int
+
+const (
+	opAnd cqlOperator = iota
+	opOr
+)
+
+var operatorMap = map[string]cqlOperator{"&": opAnd, "|": opOr}
+
+func (o *cqlOperator) Capture(s []string) error {
```

Comment:
might be nice to comment that this implements an interface from participle. the `string` methods are obvious enough, but since this comes from an outside lib, could be useful here